tests/testthat/test-classes.R

```diff
@@ -54,6 +49,7 @@ test_that("create_model_object() errors if keys are missing", {
 test_that("create_summary_object() correctly assigns class", {
   .sum <- list()
   .sum[[SUMMARY_DETAILS]] <- "naw"
+  .sum[[SUMMARY_HEURISTICS]] <- "naw"
```

Comment:
That actually doesn't work because they're variables, but that syntax converts them to strings. See here:
```
SUMMARY_DETAILS <- "run_details"
SUMMARY_HEURISTICS <- "run_heuristics"

.sum <- list(
  SUMMARY_DETAILS = "naw",
  SUMMARY_HEURISTICS = "naw"
)
str(.sum)
# List of 2
# $ SUMMARY_DETAILS   : chr "naw"
# $ SUMMARY_HEURISTICS: chr "naw"

.sum <- list()
.sum[[SUMMARY_DETAILS]] <- "naw"
.sum[[SUMMARY_HEURISTICS]] <- "naw"
str(.sum)
# List of 2
# $ run_details   : chr "naw"
# $ run_heuristics: chr "naw"
```

I also tried the dplyr `!!SUMMARY_DETAILS := "naw"` syntax as a hail mary, but it didn't work either. I don't know of any other way to pass in list element names as a variable.

Reply:
Ah, okay. `rlang::list2()` might be the answer:

```r
SUMMARY_DETAILS <- "run_details"
SUMMARY_HEURISTICS <- "run_heuristics"

rlang::list2(
  !!SUMMARY_DETAILS := "naw",
  !!SUMMARY_HEURISTICS := "naw"
)
#> $run_details
#> [1] "naw"
#> 
#> $run_heuristics
#> [1] "naw"
```